docs/requiredtools/typescript.md

```diff
@@ -7,12 +7,188 @@ parent: Required tools
+Once the package feed is accessible the following packages can be added to the project.
 
 ```Shell
+npm i @microsoft/kiota-abstractions -S 
+npm i @microsoft/kiota-authentication-azure -S
+npm i @microsoft/kiota-http-fetchlibrary -S
+npm i @microsoft/kiota-serialization-json -S
 npm i @azure/identity node-fetch
 ```
+
+Only the first package, kiota-abstractions. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.
+
+## Generating the SDK
+
+Kiota generates SDKs from OpenAPI documents. The example below is a minimal OpenAPI description that describes how to call the `/me` endpoint on Microsoft Graph.
+
+Create a file called openapi.yaml with the following contents:
+
+```yaml
+openapi: 3.0.1
+info:
+  title: Partial Graph API
+  version: v1.0
+servers:
+  - url: https://graph.microsoft.com/v1.0/
+    description: Core
+paths:
+  /me:
+    get:
+      responses:
+        200:
+          description: Success!
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/microsoft.graph.user"
+components:
+  schemas:
+    microsoft.graph.user:
+      type: object
+      properties:
+        id:
+          type: string
+        displayName:
+          type: string
+```
+
+You can then use the Kiota command line tool to generate the SDK classes.
+
+```shell
+.\kiota.exe -d openapi.yaml --language typescript -o .\src\kiota -c GraphServiceClient -n MicrosoftGraph
+```
+
+## Creating an application registration
+
+> Note: this step is required if your client will be calling APIs that are protected by the Microsoft Identity Platform like Microsoft Graph.
+
+To be able to authenticate against the demo application against Microsoft Graph, you will need to create an application registration.  You can do this via the Azure portal, or if you have [Microsoft Graph PowerShell](https://www.powershellgallery.com/packages/Microsoft.Graph) installed, you can use the following command to create the application.
+
+```PowerShell
+Connect-MgGraph -Scopes "User.Read", "Application.ReadWrite.All"
+$app = New-MgApplication -DisplayName "Kiota - NodeJS" `
+  -IsFallbackPublicClient `
+  -SignInAudience AzureADMyOrg `
+  -RequiredResourceAccess @{ "ResourceAppId" = "00000003-0000-0000-c000-000000000000"; "ResourceAccess" = @( @{ Id = "e1fe6dd8-ba31-4d61-89e7-88639da4683d"; Type = "Scope"}) }
+
+Write-Host "Your Tenant Id is : $((Get-MgOrganization).Id)"
+Write-Host "Your Client Id is : $($app.AppId)"
+Disconnect-MgGraph
+```
+
+Record the value of the ClientId property of the $app object as it will be needed in a later step.
+
+## Creating the client application
+
+Before adding our actual code, create an `auth.ts` file at the root of the `src` folder that will handle our authentication.
+
+```typescript
+import { DeviceCodeCredential } from "@azure/identity";
+import { BaseBearerTokenAuthenticationProvider } from "@microsoft/kiota-abstractions"
```

Comment:
The AzureIdentityAuthenticationProvider from the authentication package does all of that for you, you just need to pass it the credentials object from azure identity.